@ember/library-tsconfig/tsconfig.json

```diff
@@ -0,0 +1,93 @@
+    // will likely improve over time
+    "allowJs": false,
+
+    // Practically, it is *nearly* impossible to have every type-checked
+    // package in your dependency graph to have compatible types.
+    // Good stewards of the ecosystem may opt to set this to false and try to 
+    // fix packages with failures, but for most people, the error information
+    // is inactionable noise.
+    "skipLibCheck": true,
+
+    // --- TS for SemVer Types compatibility
+    // Strictness settings -- you should *not* change these: Ember code is not
+    // guaranteed to type check with these set to looser values.
+    "strict": true,
+    "noUncheckedIndexedAccess": true,
+
+    // Interop: this is viral and will require anyone downstream of your package
+    // to *also* set them to true. However, this represents the way that
+    // bundlers actually work, and is future-compatible with the closest module
+    // modes: "nodenext" in TS 4.7+ and "mixed" in 5.0+ mode. Since Ember apps
+    // and addons never emit with `tsc`, this is safe: it makes type-checking do
+    // the right thing, but does not result in changes to what gets emitted. We
+    // intentionally leave `esModuleInterop` unset, so that it gets whatever TS
+    // provides as the default for the currently-specified `module` mode.
+    "allowSyntheticDefaultImports": true,
+
+    // --- Lint-style rules
+
+    // TypeScript also supplies some lint-style checks; nearly all of them are
+    // better handled by ESLint with the `@typescript-eslint`. This one is more
+    // like a safety check, though, so we leave it on.
+    "noPropertyAccessFromIndexSignature": true,
+
+    // --- Compilation/integration settings
+    // Setting `noEmitOnError` here allows tools trying to respect the tsconfig
+    // to still emit code without breaking on errors.
+    // Errors are still reported in the CLI when running `tsc` or `glint`,
+    // but the errors won't prevent code from being emitted.
+    // This helps hasten development by allowing devs to prototype before coming
+    // to a decision on what they want their types to be.
+    "noEmitOnError": false,
+
+    // We use Babel for emitting runtime code, because it's very important that
+    // we always and only use the same transpiler for non-stable features, in
+    // particular decorators. If you were to change this to `true`, it could
+    // lead to accidentally generating code with `tsc` instead of Babel, and
+    // could thereby result in broken code at runtime.
+    "noEmit": true,
+
+    // Ember makes heavy use of decorators; TS does not support them at all
+    // without this flag.
+    "experimentalDecorators": true,
+
+    // We don't use TS for compilation, so we can disable these.
+    // Library authors should set declaration and declarationMap to true, however
+    "declaration": true,
+    "declarationMap": true,
+
+    // We don't compile with TS
+    "inlineSourceMap": false,
+    "inlineSources": false,
```

Comment:
Don't we need `"inlineSources": true` for source maps to actually work? Otherwise addons have to publish their `src` directory since the paths will be `../src/whatever.ts`?

Reply:
I just tested this and there is no change in .d.ts.map, true or false. likely because I'm not using tsc to build the actual sources.

Reply:
No, it would be in the `.map` files in `dist`

Reply:
Which, at least in my setup which I think is based on a pretty recent v2 addon blueprint, are generated by rollup/babel/tsc, so are governed by these compiler options

Reply:
I don't see any difference in the dist's .map files either.

Reply:
Huh, something in the rollup/babel pipeline must be enabling inline source then 🤔

Reply:
what do you mean? do you have a library I should look at that generates big .d.ts.map files?

Reply:
Sorry, to be clear, I no longer think we need changes to this PR.

I'm just noticing that I think our rollup/babel pipeline enable `sourceMaps` and `inlineSourceMaps` regardless of `tsconfig.json`. In a project using just `tsc` you'd need those options enabled in `tsconfig.json` to get usable-by-library-consumers source maps.

Reply:
rogerroger.gif

good thing about us managing our tsconfigs now is that we can do patch releases when we need to, and majors when we need to :tada: